[BUG]  [DB 14.3] `numRecords` stat in Delta Lake tables is not set on GPU

**Describe the bug**
When running the test `delta_lake_update_test.py::test_delta_update_dataframe_api` it fails with the following exception 

```
E               AssertionError: Delta log 00000000000000000002.json 'remove' keys mismatch:
E               CPU: {'path': 'part-00000-52012349-499d-46fa-bec2-86be7b58c965-c000.snappy.parquet', 'deletionTimestamp': 1738197316431, 'dataChange': True, 'extendedFileMetadata': True, 'partitionValues': {}, 'size': 97114, 'tags': {'OPTIMIZE_TARGET_SIZE': '268435456'}, 'stats': '{"numRecords":2048}'}
E               GPU: {'path': 'part-00000-e1fb51cc-00b6-4f0a-9ba6-a57828a91906-c000.snappy.parquet', 'deletionTimestamp': 1738197322059, 'dataChange': True, 'extendedFileMetadata': True, 'partitionValues': {}, 'size': 97114, 'tags': {'OPTIMIZE_TARGET_SIZE': '268435456'}}

```

**Steps/Code to reproduce bug**
Build spark-rapids for Databricks 14.3 and run the command `TEST_PARALLEL=0 TEST_MODE=DELTA_LAKE_ONLY WITH_DEFAULT_UPSTREAM_SHIM=0 TESTS=delta_lake_update_test.py::test_delta_update_dataframe_api ./jenkins/databricks/test.sh`

**Expected behavior**
The test should pass


**Additional context**
I think we need to update the GpuUpdateCommand.scala for spark350db143 with the new stats

--- a/delta-lake/delta-spark350db143/src/main/scala/com/databricks/sql/transaction/tahoe/rapids/GpuUpdateCommand.scala
+++ b/delta-lake/delta-spark350db143/src/main/scala/com/databricks/sql/transaction/tahoe/rapids/GpuUpdateCommand.scala
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024, NVIDIA CORPORATION.
+ * Copyright (c) 2024-2025, NVIDIA CORPORATION.
  *
  * This file was derived from UpdateCommand.scala
  * in the Delta Lake project at https://github.com/delta-io/delta.
@@ -21,9 +21,8 @@
 
 package com.databricks.sql.transaction.tahoe.rapids
 
-import com.databricks.sql.transaction.tahoe.{DeltaLog, DeltaOperations, DeltaTableUtils, DeltaUDF, OptimisticTransaction}
+import com.databricks.sql.transaction.tahoe.{DeltaLog, DeltaOperations, DeltaTableUtils, DeltaUDF, OptimisticTransaction, RowTracking}
 import com.databricks.sql.transaction.tahoe.DeltaCommitTag._
-import com.databricks.sql.transaction.tahoe.RowTracking
 import com.databricks.sql.transaction.tahoe.actions.{AddCDCFile, AddFile, FileAction}
 import com.databricks.sql.transaction.tahoe.commands.{DeltaCommand, DMLUtils, UpdateCommand, UpdateMetric}
 import com.databricks.sql.transaction.tahoe.files.{TahoeBatchFileIndex, TahoeFileIndex}
@@ -109,7 +108,8 @@ case class GpuUpdateCommand(
     val (metadataPredicates, dataPredicates) =
       DeltaTableUtils.splitMetadataAndDataPredicates(
         updateCondition, txn.metadata.partitionColumns, sparkSession)
-    val candidateFiles = txn.filterFiles(metadataPredicates ++ dataPredicates)
+    val candidateFiles = txn.filterFiles(metadataPredicates ++ dataPredicates,
+      keepNumRecords = true /* Keep numRecords regardless of the state of deletion vectors in DB */)
     val nameToAddFile = generateCandidateFileMap(deltaLog.dataPath, candidateFiles)
 
     scanTimeMs = (System.nanoTime() - startTime) / 1000 / 1000

--- a/integration_tests/src/main/python/delta_lake_update_test.py
+++ b/integration_tests/src/main/python/delta_lake_update_test.py
@@ -192,7 +192,6 @@ def test_delta_update_rows_with_dv(spark_tmp_path, use_cdf, partition_columns, e
 @datagen_overrides(seed=0, reason='https://github.com/NVIDIA/spark-rapids/issues/10025')
 @pytest.mark.parametrize("enable_deletion_vector", deletion_vector_values_with_350DB143_xfail_reasons(
                             enabled_xfail_reason='https://github.com/NVIDIA/spark-rapids/issues/12042'), ids=idfn)
-@pytest.mark.xfail(condition=is_databricks143_or_later(), reason="https://github.com/NVIDIA/spark-rapids/issues/12047")
 def test_delta_update_dataframe_api(spark_tmp_path, use_cdf, partition_columns, enable_deletion_vector):
     from delta.tables import DeltaTable
     data_path = spark_tmp_path + "/DELTA_DATA"

